test/integration/statefulset/statefulset_test.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package statefulset
+
+import (
+	"testing"
+
+	"k8s.io/api/apps/v1beta1"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/kubernetes/test/integration/framework"
+)
+
+func createSTSsPods(t *testing.T, clientSet clientset.Interface, stss []*v1beta1.StatefulSet, pods []*v1.Pod, ns string) ([]*v1beta1.StatefulSet, []*v1.Pod) {
+	stsClient := clientSet.AppsV1beta1().StatefulSets(ns)
+	podClient := clientSet.Core().Pods(ns)
+	var createdSTSs []*v1beta1.StatefulSet
+	var createdPods []*v1.Pod
+	for _, sts := range stss {
+		createdSTS, err := stsClient.Create(sts)
+		if err != nil {
+			t.Fatalf("Failed to create sts %s: %v", sts.Name, err)
+		}
+		createdSTSs = append(createdSTSs, createdSTS)
+	}
+	for _, pod := range pods {
+		createdPod, err := podClient.Create(pod)
+		if err != nil {
+			t.Fatalf("Failed to create pod %s: %v", pod.Name, err)
+		}
+		createdPods = append(createdPods, createdPod)
+	}
+
+	return createdSTSs, createdPods
+}
+
+// Verify .Status.Replicas is equal to .Spec.Replicas
+func waitSTSStable(t *testing.T, clientSet clientset.Interface, sts *v1beta1.StatefulSet, ns string) {
+	stsClient := clientSet.AppsV1beta1().StatefulSets(ns)
+	if err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+		newSTS, err := stsClient.Get(sts.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		if newSTS.Status.Replicas == *newSTS.Spec.Replicas {
+			return true, nil
+		}
+		return false, nil
+	}); err != nil {
+		t.Fatalf("Failed to verify .Status.Replicas is equal to .Spec.Replicas for sts %s: %v", sts.Name, err)
+	}
+}
+
+func testScalingUsingScaleSubresource(t *testing.T, c clientset.Interface, sts *v1beta1.StatefulSet, ns string, x int32) {
+	kind := "StatefulSet"
+	scaleClient := c.ExtensionsV1beta1().Scales(ns)
+	if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		scale, err := scaleClient.Get(kind, sts.Name)
```

Comment:
This fails with the error `the server could not find the requested resource`. The `/scale` subresource may not be registered for `apps/v1beta1` statefulset.